Modules/Module_Processing/xASL_init_LoadDataPar.m

```diff
@@ -134,4 +136,61 @@
     %% Load dataPar
     x = xASL_adm_MergeStructs(dataPar.x, x);
     
-end
\ No newline at end of file
+end
+
+%% -----------------------------------------------------------------------------
+%% -----------------------------------------------------------------------------
+function [x] = xASL_initLoadDataPar_PrepareAtlas4ROI(x)
+%Check for x.S.Atlases & x.S.TissueMasking and provide default values or print instructions
+
+bAtlasTissueMatch = true; % Necessary condition
+
+if ~isfield(x, 'S') || (~isfield(x.S,'Atlases') && ~isfield(x.S, 'TissueMasking'))
+	% Default atlases/ROIs & tissue masks if nothing is provided
+	x.S.Atlases = {'TotalGM','DeepWM'}; % Default
+    x.S.TissueMasking = {'GM' 'WM'}; % GM WM, fits with the TotalGM & DeepWM above
+    % Note that this should be in the same order as the atlases/ROIs
+    % A mismatch (e.g. TissueMasking=GM for Atlases=deepWM) would result in an empty ROI, producing a NaN in the .tsv table
+elseif ~isfield(x.S, 'Atlases') && isfield(x.S, 'TissueMasking')
+	% Missing Atlases, but provided TissueMasking - cannot continue
+    warning('Custom tissue-types (x.S.TissueMasking) specified without ROI atlas-selection (x.S.Atlases). Atlases need to be provided. See instructions below:');
+    bAtlasTissueMatch = false;
+elseif isfield(x.S, 'Atlases') && isfield(x.S, 'TissueMasking') && length(x.S.Atlases)~=length(x.S.TissueMasking)
+	% Non matching lengths, cannot continue
+    warning('The same number of ROI atlases as subject-wise tissue-types provided does not match:');
```

Comment:
Should be:

warning('The number of ROI atlases `x.S.Atlases` as subject-wise tissue-types `x.S.TissueMasking` provided does not match:');

Reply:
Good catch. Fixed.